model/task_lifecycle_test.go

```diff
@@ -6319,7 +6319,7 @@ func TestEvalStepbackDeactivatePrevious(t *testing.T) {
 	finishedTask := task.Task{
 		Id:                  "t3",
 		BuildId:             "b3",
-		Status:              evergreen.TaskUndispatched,
+		Status:              evergreen.TaskSucceeded,
```

Comment:
Why does this test check a different status? Does the (linear) stepback logic behave differently now?

Reply:
I'm not sure why originally the status here was set as `TaskUndispatched` but the test later gives evalStepback `TaskSucceeded`- which causes a mismatch of the statuses.

The test is testing if a linear stepback iteration succeeded it should deactivate the previous one.

The reason this wasn't previously erroring before is that we would pass what evalStepback gets in to linear stepback- but now linear stepback just reads it off of the task (L1304).

i.e. it was a test data inconsistency